### PR TITLE
Move gotoc tests under run-make to their own test suite.

### PR DESCRIPTION
**Pull request by bdalrhm**
### Description of changes: 

This PR moves `gotoc*` tests under `run-make` test suite to their own test suite, `expected`. It removes the scripts used to run those tests because the logic for running the tests is now handled by compiletest.

### Resolved issues:

None.

### Testing:

* How is this change tested?
  * The moved tests still run
* Is this a refactor change?
  * Yes. `gotoc*` tests under `src/test/run-make` were moved to `src/test/expected`

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -2434,7 +2435,33 @@ impl<'test> TestCx<'test> {
...
+    fn run_expected_test(&self) {
+        // We create our own command for the same reasons listed in `run_rmc_test` method.
+        let mut rmc = Command::new("rmc");
+        // Pass the test path along with RMC and CBMC flags parsed from comments at the top of the test file.
+        rmc.args(&self.props.rmc_flags)
+            .arg(&self.testpaths.file)
+            .arg("--")
+            .args(&self.props.cbmc_flags);
+        self.add_rmc_dir_to_path(&mut rmc);
+        self.add_rmc_dir_to_path(&mut rmc);
```
Comment (adpaco-aws) [inline src/tools/compiletest/src/runtest.rs:2460]: Duplicated instruction.

File: src/tools/compiletest/src/runtest.rs [hunk 2]
```diff
@@ -2434,7 +2435,33 @@ impl<'test> TestCx<'test> {
         self.add_rmc_dir_to_path(&mut cargo);
         let proc_res = self.compose_and_run_compiler(cargo, None);
         let expected = fs::read_to_string(self.testpaths.file.clone()).unwrap();
-        // Print an error if the verification result does not contains the expected lines.
+        // Print an error if the verification result does not contain the expected lines.
```
Comment (adpaco-aws) [inline src/tools/compiletest/src/runtest.rs:2438]:
```suggestion
        // Print an error if the verification output does not contain the expected lines.
```

File: src/tools/compiletest/src/runtest.rs [hunk 3]
```diff
@@ -2434,7 +2435,33 @@ impl<'test> TestCx<'test> {
...
+        if let Some(line) = TestCx::contains_lines(&proc_res.stdout, expected.split('\n').collect())
+        {
+            self.fatal_proc_rec(
+                &format!("test failed: expected output to contain the line: {}", line),
+                &proc_res,
+            );
+        }
```
Comment (adpaco-aws) [inline src/tools/compiletest/src/runtest.rs:2445]: The same code is used for `cargo-rmc` and `expected` so we can refactor into a function.

File: scripts/rmc-regression.sh [hunk 4]
```diff
@@ -21,10 +19,6 @@ check-cbmc-viewer-version.py --major 2 --minor 5
...
 
-# Standalone rmc tests and cargo tests
-pushd $RUST_DIR
+# Standalone rmc tests, cargo tests, and expected tests
 ./x.py build -i --stage 1 library/std ${EXTRA_X_PY_BUILD_ARGS}
-./x.py test -i --stage 1 cbmc firecracker prusti smack cargo-rmc
-
-# run-make tests
-./x.py test -i --stage 1 src/test/run-make --test-args gotoc
+./x.py test -i --stage 1 cbmc firecracker prusti smack cargo-rmc expected
```
Comment (adpaco-aws) [inline scripts/rmc-regression.sh:24]: 💯

File: src/tools/compiletest/src/runtest.rs [hunk 5]
```diff
@@ -2434,7 +2435,33 @@ impl<'test> TestCx<'test> {
...
+    }
+
+    /// Runs RMC on the test file specified by `self.testpaths.file`. An error
+    /// message is printed to stdout if verification result does not contain
```
Comment (adpaco-aws) [inline src/tools/compiletest/src/runtest.rs:2449]:
```suggestion
    /// message is printed to stdout if verification output does not contain
```

**Review by adpaco-aws — CHANGES_REQUESTED** [review]

File: src/tools/compiletest/src/runtest.rs (hunk 1 above)
Comment (bdalrhm) [reply src/tools/compiletest/src/runtest.rs]: Nice catch!

File: src/tools/compiletest/src/runtest.rs (hunk 3 above)
Comment (bdalrhm) [reply src/tools/compiletest/src/runtest.rs]: Done!

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**